internal/configs/config.go

```diff
@@ -484,7 +484,14 @@ func (c *Config) addProviderRequirements(reqs getproviders.Requirements, recurse
 					})
 					continue
 				}
-				if i.ProviderConfigRef.Name != target.ProviderConfigRef.Name || i.ProviderConfigRef.Alias != target.ProviderConfigRef.GetNoKeyAlias() {
+
+				// We want to compare aliases only if there is no `each` or `count` inside a resource provider reference.
+				haveIncompatibleAliases := true
+				if !target.ProviderConfigRef.HasInstanceRefsInAlias() {
```

Comment:
I wonder if a pattern like `alias, ok := target.ref.KnownAlias(); ok` would make more sense here?  I also stink at naming functions.

Reply:
Given that import blocks are only allowed in the root module, we should know the exact provider alias and can treat any unknowns here as a bug?

Reply:
We treat it as an error because the provider could be set in both resource and import blocks. If that is the case, providers (and aliases) should match. If the provider alias contains `each` / `count`, they will be different for sure so we error. Does it answer your question?